app/assets/javascripts/admin/forms/project_step_form.js.coffee

```diff
@@ -0,0 +1,151 @@
+              parent.html($(html).html())
+              setLinksVisibility(parent)
+              setCallbacks(parent)
+              )
+            .on "ajax:error", (e, xhr, status, error) ->
+              $(scope).find(".error_notification").show()
+
+    #
+    # ajax errors notification
+    #
+    $("[data-hide]").click (evt) ->
+      $(scope).find(".error_notification").hide()
+
+    #
+    # remove language
+    #
+
+    $(scope).find('.remove_language').click (evt) ->
+      evt.preventDefault()
+
+      locale = $(evt.target).data('locale')
+      $(scope).find(".step-form-left-group[data-locale=\"#{locale}\"]").remove()
+      addToAvailableLocales(scope, locale)
+
+      deleted_locales = JSON.parse($("#deleted_locales").val())
+      deleted_locales.push(locale)
+      $(scope).find("#deleted_locales").val(JSON.stringify(deleted_locales))
+      setLinksVisibility(scope)
+
+    #  
+    # add another language
+    #
+
+    $(scope).find('#add_another_language').click (evt) ->
+      evt.preventDefault()
+
+      newLocale = getNextAvailableLocale(scope)
+
+
+      # clone the last language block and set it up with the next available locale
+      lastLanguageBlock = $(scope).find(".step-form-left-group").last()
+      clone = lastLanguageBlock.clone()
+      localeToReplace = $(clone).data('locale')
+      html = $(clone).prop('outerHTML')
+      html = html.replace(new RegExp("_#{localeToReplace}", 'g'), "_#{newLocale}")
+                 .replace(new RegExp("data-locale=\"#{localeToReplace}\"", 'g'), "data-locale=\"#{newLocale}\"")
+      newLanguageBlock = $.parseHTML(html)
+      $(newLanguageBlock).data('locale', newLocale)
+      $(newLanguageBlock).find(".summary").attr("placeholder", gon.I18n[newLocale].summary).val('')
+      $(newLanguageBlock).find(".details").attr("placeholder", gon.I18n[newLocale].details).val('')
+      $(newLanguageBlock).find("#project_step_locale_#{newLocale} option[value='#{newLocale}']").attr("selected", "selected")
+      $(newLanguageBlock).find("#remove_#{newLocale}_language").click (evt) ->
+        evt.preventDefault()
+        $(scope).find(".step-form-left-group[data-locale=\"#{newLocale}\"]").remove()
+        addToAvailableLocales(scope, newLocale)
+        setLinksVisibility(scope)
+
+      $(newLanguageBlock).insertAfter(lastLanguageBlock)
+      setLinksVisibility(scope)
+      setCallbacks(scope)
+
+    #
+    # Change existing translation locale
+    #
+
+    $(scope).find('.locale').change (evt) ->
+      evt.preventDefault()
+      newLocale = $(evt.target).val()
+      parent = $(evt.target).parents(".language-block")
+      parent.find('.summary').attr("placeholder", gon.I18n[newLocale].summary);
+      parent.find('.details').attr("placeholder", gon.I18n[newLocale].summary);
+
+  #
+  # Run once on load
+  #
+
+
+  $('div .panel.step').each((idx, el) ->
+    setLinksVisibility(el)
+    setCallbacks(el)
+    initTypeSelects(el)
+  )
+
+
```

Comment:
Hi Tom.  Thanks much for the additional code review feedback and update about moving forward with Backbone!  Just let me know when things are ready for me to merge the latest from develop and proceed.